gamla/functional_generic.py

```diff
@@ -130,6 +131,12 @@ def compose(*funcs):
     #         logging.error(f"composing {prev} after {f}, which don't fit")
     #     prev = f
 
+    if operator.empty(funcs):
+        return gamla.identity
+
+    if len(funcs) == 1:
+        return operator.head(funcs)
```

Comment:
@nitzo remember this is a very sensitive place performance wise, probably funcs[0] is faster than an additional function call
similarly `not funcs` will be faster than `operator.empty(funcs)`